src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -1821,6 +1829,48 @@ public static boolean fromTheSameCaseExpression(@NotNull PsiElement origin, @Not
     return false;
   }
 
+  public static class ErlangFunctionCallNamedParameter<T extends PsiElement> extends PatternCondition<T> {
+    private final List<String> NAMES;
+
+    public ErlangFunctionCallNamedParameter(@NotNull String... names) {
+      super("functionCallNamedParameter");
+      NAMES = ContainerUtil.newArrayList(names);
+    }
+
+    @Override
+    public boolean accepts(@NotNull T element, ProcessingContext context) {
+      ErlangQAtom argument = element instanceof ErlangQAtom ? (ErlangQAtom) element : null;
+      if (argument == null) return false;
+
+      ErlangFunctionCallExpression call = PsiTreeUtil.getParentOfType(argument, ErlangFunctionCallExpression.class);
+      ErlangFunction function = resolveToFunction(call);
+      if (function == null) return false;
+
+      int position = getPositionOfArgument(argument);
+      if (position == -1) return false;
+
+      for (ErlangFunctionClause clause : function.getFunctionClauseList()) {
+        List<ErlangArgumentDefinition> definitions = clause.getArgumentDefinitionList().getArgumentDefinitionList();
+        ErlangArgumentDefinition definition = definitions.size() > position ? definitions.get(position) : null;
+        ErlangQVar var = PsiTreeUtil.findChildOfType(definition, ErlangQVar.class);
+        if (var != null && NAMES.contains(var.getName())) {
+          return true;
+        }
+      }
+
+      return false;
+    }
+
+    private static int getPositionOfArgument(@NotNull ErlangQAtom argument) {
+      ErlangArgumentList argumentList = PsiTreeUtil.getParentOfType(argument, ErlangArgumentList.class);
+      ErlangExpression expression = PsiTreeUtil.getParentOfType(argument, ErlangExpression.class, false);
+      if (expression != null && expression.getParent() instanceof ErlangColonQualifiedExpression) {
+        expression = (ErlangExpression) expression.getParent();
+      }
```

Comment:
Let me explain. When we pass a function as parameter, there are 2 scenarios: `my_fun(foo)` or `my_fun(main:foo)`.
So I don't need to cast it to ErlangColonQualifiedExpression. I just need to get one level up, if it's colon qualified. If not - I need to do nothing.